### ISSUE TITLE
input shape for training

### ISSUE THREAD

**Issue by shuyuan-wang** [ultralytics/ultralytics#19525]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

did you force the input shape of training to be square? I put rect=True and imgsz=[960,540], the input shape for training is still [960,960]

**Comment by UltralyticsAssistant** [comment]
👋 Hello @shuyuan-wang, thank you for your interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com/) for detailed insights, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples that might address your question on input shapes for training.

If this is a custom training ❓ Question, please share more details about your training setup, including your dataset specifics, model configuration, and relevant logs to help us provide a clearer response. Also, check if you’re following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

If this turns out to be a 🐛 Bug Report, please include a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) that showcases the issue so we can assist effectively.

To ensure you are using the latest features and fixes, upgrade your `ultralytics` package along with all dependencies in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/) using the command below:

```bash
pip install -U ultralytics
```

## Community Support and Resources

💬 Join the Ultralytics community!
- Real-time discussions: [Discord](https://discord.com/invite/ultralytics) 🎧 
- In-depth conversations: [Discourse](https://community.ultralytics.com/)
- Broader sharing: [Subreddit](https://reddit.com/r/Ultralytics)

## Notebooks and Verified Environments

YOLO runs seamlessly in multiple environments. Here are some easy ways to get started:

- **Free GPU Notebooks**: 
  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> 
  <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> 
  <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **On Cloud**: Use [Google Cloud](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/) or [AWS](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/).
- **Docker Image**: Refer to our [Docker Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) and download directly <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>.

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing. CI tests validate all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) daily on macOS, Windows, and Ubuntu.

This is an automated response to help guide you 🔍. An Ultralytics engineer will assist you further soon. Let us know if you need more clarification! 🙂

**Comment by Y-T-G** [comment]
`rect=True` uses minimum padding. But if you have square images in your dataset, then it will pad to square to match that.
